### PR TITLE
Several minor code quality improvements + DG update

### PR DESCRIPTION
**Pull request by JoeyChenSmart**
Mostly adding comments/JavaDocs.

Also update DG to reflect updated macro component.

### CODE REVIEW

File: src/main/java/jimmy/mcgymmy/logic/parser/parameter/Parameter.java [hunk 1]
```diff
@@ -14,7 +14,7 @@
     private Optional<T> value = Optional.empty();
 
     /**
-     * Create a required parameter
+     * Creates a required parameter
```
Comment (Jh123x) [inline src/main/java/jimmy/mcgymmy/logic/parser/parameter/Parameter.java]: Missing Fullstop?

File: src/main/java/jimmy/mcgymmy/logic/parser/parameter/OptionalParameter.java [hunk 2]
```diff
@@ -14,7 +14,7 @@
     private Optional<T> value = Optional.empty();
 
     /**
-     * Create an optional parameter
+     * Creates an optional parameter
```
Comment (Jh123x) [inline src/main/java/jimmy/mcgymmy/logic/parser/parameter/OptionalParameter.java]: Same here

File: docs/AboutUs.md [hunk 3]
```diff
@@ -26,7 +26,7 @@ You can reach us at the email `seer[at]comp.nus.edu.sg`
 <img src="images/chewypiano.png" width="200px">
 
 [github](http://github.com/chewypiano)
-[portfolio](team/kamil.md)
+[portfolio](team/chewypiano)
```
Comment (Jh123x) [inline docs/AboutUs.md]: Missing .html at the back for some of them (Edit: Oops its not .md >< Might be .HTML)

File: src/main/java/jimmy/mcgymmy/logic/parser/McGymmyParser.java [hunk 4]
```diff
@@ -48,14 +53,15 @@ public McGymmyParser(MacroList macroList) {
      * @throws ParseException if an argument to the command is not in the correct format
      */
     public CommandExecutable parse(String text) throws ParseException {
-        ParserUtil.HeadTailString headTail = ParserUtil.HeadTailString.splitString(text);
-        String commandName = headTail.getHead();
+        Pair<String, String[]> headTail = ParserUtil.splitString(text);
+        String commandName = headTail.getKey();
+        String[] commandArguments = headTail.getValue();
```
Comment (Jh123x) [inline src/main/java/jimmy/mcgymmy/logic/parser/McGymmyParser.java]: Might want to change the name to a Pair but it is a minor issue ><

**Review by Jh123x — CHANGES_REQUESTED** [review]
Almost LGTM Minor Changes required x.x

File: docs/AboutUs.md (hunk 3 above)
Comment (JoeyChenSmart) [reply docs/AboutUs.md]: this is the correct format. Just checked on my fork and this gives the correct link

**Review by Jh123x — APPROVED** [review]
LGTM

**Pull request merged**